Configurations lacking explicit artifacts should still resolve

tresat opened the pull request:
Supercedes #29753 and fixes #29630.

Stop making the primary variant unselectable when a consumable configuration has secondary variants but no artifacts on its primary. The primary now stays in the candidate set during variant selection.

To preserve the previous outcome for existing consumers, Gradle auto-tags such configurations with a new `org.gradle.fallback-variant` attribute (primary = true, each secondary = false) and registers a default disambiguation rule that prefers false over true. Net effect for silent consumers: the secondary still wins, exactly as before. Net effect for consumers that want the empty primary: they can now opt in by requesting org.gradle.fallback-variant=true on the resolvable configuration — an escape hatch that didn't exist before.

Changes

- New public API `org.gradle.api.attributes.FallbackVariant` (@Incubating, @since 9.7.0) with TRUE/FALSE value constants and the FALLBACK_VARIANT_ATTRIBUTE identifier.
- DefaultConfigurationPublications augments primary and secondary variant attributes via addFallbackIfNecessary when the configuration is "fallback-shaped" (primary has no statically-declared artifacts AND secondaries exist). Respects user-supplied values: if a build author already set the attribute on a variant, it is left untouched.

Limitations

The "no artifacts" check uses PublishArtifactSet.isEmpty() at variant-computation time. A primary that declares a lazy artifact provider which later resolves to an empty file set is NOT treated as a fallback — the lazy declaration counts as non-empty.

tresat committed:
Setup reproducers to walk through scenarios

tresat committed:
Setup reproducers to walk through scenarios

# Conflicts:
#	platforms/software/dependency-management/src/integTest/groovy/org/gradle/integtests/experiments/SecondaryVariantsBreakConsumableConfs.groovy

tresat committed:
Expand and simplify test cases, demo other reports

tresat committed:
Expand and simplify test cases, demo other reports

tresat committed:
Always visit your own variant first, even if you have children (secondary variants)

tresat committed:
Always visit your own variant first, even if you have children (secondary variants)

tresat committed:
Always add primary variant, warn on primary variant without artifacts with secondary variant.

tresat committed:
Adjust test expectations for new deprecation warning and continued selectability of the primary variant

tresat committed:
Update test expectations

tresat committed:
Fix tests for CC compatibility

tresat committed:
Remove deprecation for primary variant with no artifacts with secondary

tresat commented:
@bot-gradle test this

bot-gradle commented:
<!-- {"replyTargetCommentId":2296760618,"teamCityBuildId":null,"teamCityBuildHeadRef":"a1844727c92f12573021248e1ba3d8b2e6f573c0","teamCityBuildStatus":"PENDING","replyApproveReviewId":null} -->
Sorry some internal error occurs, please contact the administrator @gradle/bt-developer-productivity

tresat commented:
@bot-gradle test this

bot-gradle commented:
<!-- {"replyTargetCommentId":null,"teamCityBuildId":null,"teamCityBuildHeadRef":"a1844727c92f12573021248e1ba3d8b2e6f573c0","teamCityBuildStatus":"PENDING","replyApproveReviewId":null} -->
I've triggered the following builds for you. Click [here](https://ge.gradle.org/scans?search.buildOutcome=failure&amp;search.names=gitCommitId&amp;search.tags=CI&amp;search.values=a1844727c92f12573021248e1ba3d8b2e6f573c0) to see all build failures.
- [Pull Request Feedback (Trigger) (Check)](https://builds.gradle.org/build/87361966)

tresat committed:
Fix test expectations

tresat commented:
@bot-gradle test this

bot-gradle commented:
<!-- {"replyTargetCommentId":null,"teamCityBuildId":null,"teamCityBuildHeadRef":"a193f7c3c3989cde84cb939ab8b37f4127aca594","teamCityBuildStatus":"PENDING","replyApproveReviewId":null} -->
I've triggered the following builds for you. Click [here](https://ge.gradle.org/scans?search.buildOutcome=failure&amp;search.names=gitCommitId&amp;search.tags=CI&amp;search.values=a193f7c3c3989cde84cb939ab8b37f4127aca594) to see all build failures.
- [Pull Request Feedback (Trigger) (Check)](https://builds.gradle.org/build/87377278)

tresat commented:
@bot-gradle test this

bot-gradle commented:
<!-- {"replyTargetCommentId":null,"teamCityBuildId":null,"teamCityBuildHeadRef":"06856c0d99b6d39a8dc3f7564d78cc2199784548","teamCityBuildStatus":"PENDING","replyApproveReviewId":null} -->
I've triggered the following builds for you. Click [here](https://ge.gradle.org/scans?search.buildOutcome=failure&amp;search.names=gitCommitId&amp;search.tags=CI&amp;search.values=06856c0d99b6d39a8dc3f7564d78cc2199784548) to see all build failures.
- [Pull Request Feedback (Trigger) (Check)](https://builds.gradle.org/build/87468702)

tresat commented:
Unstale

tresat commented:
Not stale

lkasso commented on platforms/documentation/docs/src/docs/userguide/releases/upgrading/upgrading_version_8.adoc:
```diff
@@ -147,6 +147,16 @@ When this happens, configurations in the substituted project could be selected t
 Consuming non-consumable configurations in this manner is deprecated and will result in an error in Gradle 9.0.
 
+[[variants_with_no_artifacts]]
+=== Defining secondary variants for primary variants does not remove primary variants
```
```suggestion
==== Primary variants without artifacts are no longer removed when secondary variants are present
```

lkasso commented on platforms/documentation/docs/src/docs/userguide/releases/upgrading/upgrading_version_8.adoc:
```diff
@@ -147,6 +147,16 @@ When this happens, configurations in the substituted project could be selected t
+[[variants_with_no_artifacts]]
+=== Defining secondary variants for primary variants does not remove primary variants
+
+It is possible to define [Secondary Variants](https://docs.gradle.org/8.9/userguide/publishing_customization.html#sec:publishing-custom-components) for consumable "primary" variants that lack artifacts.
```
```suggestion
You can define [secondary variants](https://docs.gradle.org/8.9/userguide/publishing_customization.html#sec:publishing-custom-components) for a consumable primary variant that has no artifacts.
```

lkasso commented on platforms/documentation/docs/src/docs/userguide/releases/upgrading/upgrading_version_8.adoc:
```diff
@@ -147,6 +147,16 @@ When this happens, configurations in the substituted project could be selected t
+
+It is possible to define [Secondary Variants](https://docs.gradle.org/8.9/userguide/publishing_customization.html#sec:publishing-custom-components) for consumable "primary" variants that lack artifacts.
+
+Defining secondary variants for a primary variant lacking artifacts used to result in the primary variant being removed from consideration during variant selection.
```
```suggestion
In earlier versions of Gradle, doing so would cause the primary variant to be silently excluded from variant selection, because it lacked artifacts.
```

lkasso commented on platforms/documentation/docs/src/docs/userguide/releases/upgrading/upgrading_version_8.adoc:
```diff
@@ -147,6 +147,16 @@ When this happens, configurations in the substituted project could be selected t
+It is possible to define [Secondary Variants](https://docs.gradle.org/8.9/userguide/publishing_customization.html#sec:publishing-custom-components) for consumable "primary" variants that lack artifacts.
+
+Defining secondary variants for a primary variant lacking artifacts used to result in the primary variant being removed from consideration during variant selection.
+Now such primary variants remain selectable, even if they lack artifacts.
```
```suggestion
Starting with Gradle 9.0, such primary variants are now considered during variant selection, even if they have no artifacts.
```

lkasso commented on platforms/documentation/docs/src/docs/userguide/releases/upgrading/upgrading_version_8.adoc:
```diff
@@ -147,6 +147,16 @@ When this happens, configurations in the substituted project could be selected t
+Defining secondary variants for a primary variant lacking artifacts used to result in the primary variant being removed from consideration during variant selection.
+Now such primary variants remain selectable, even if they lack artifacts.
+
+In very rare cases, when a variant request's attributes are underspecified, this change may cause a different variant to be selected than in previous versions of Gradle.
```
```suggestion
NOTE: In rare cases, when the variant selection request is underspecified, this change may cause Gradle to select a different variant than it did before.
```

tresat commented:
We need to get this ready for AGP to test with.

tresat committed:
Fix tests and update release notes

lkasso reviewed (approved):
Docs LGTM but you may want to point to variant_aware_resolution.html#sec:secondary-variants as well

tresat committed:
Relocate variant artifacts note to Gradle 9.7

Move the release note regarding primary variants without artifacts from the Gradle 9.0 upgrade section to the Gradle 9.7 upgrade section. This aligns the documentation with the updated target version for this behavior change.

tresat committed:
Standardize variant name display and update docs

Ensures consistent quoting for variant names in test output messages and enhances related documentation with a cross-reference to secondary variants.

tresat committed:
Tweak upgrade guide

tresat commented:
@bot-gradle test RFR

bot-gradle commented:
<!-- {"replyTargetCommentId":4625713874,"triggeredTeamCityBuilds":[{"buildConfigurationId":"Gradle_Master_Check_Stage_ReadyforRelease_Trigger","buildId":"113494047","headRef":"5c28b1bf23d1d713066ffb52fd7ebaf1cf9ae92c"}]} -->
I've triggered the following builds:
- [Ready for Release (Trigger) (Check)](https://builds.gradle.org/build/113494047)
  - [Build Scan](https://ge.gradle.org/scans?search.tags=CI&amp;search.names=gitCommitId,PartOf&amp;search.values=5c28b1bf23d1d713066ffb52fd7ebaf1cf9ae92c,ReadyforRelease&amp;)

bot-gradle commented:
The following builds have failed:
- [Ready for Release (Trigger) (Check)](https://builds.gradle.org/buildConfiguration/Gradle_Master_Check_Stage_ReadyforRelease_Trigger/113494047)
  - [Build Scan](https://ge.gradle.org/scans?search.buildOutcome=failure&amp;search.tags=CI&amp;search.names=gitCommitId&amp;search.values=5c28b1bf23d1d713066ffb52fd7ebaf1cf9ae92c&amp;)

tresat committed:
Correct versions in comment

tresat committed:
Introduce fallback attribute to prefer secondary variants

Gradle 10.0 now considers primary variants without artifacts during
selection. To maintain the behavior of preferring secondary variants
when they are present, a new `org.gradle.fallback-variant` attribute
is automatically applied. The primary variant is tagged with `true`
and secondary variants with `false`. A default disambiguation rule
prefers variants tagged `false`, ensuring secondaries are selected
by default. Consumers can explicitly request `fallback-variant=true`
to opt into the empty primary.

Fixes #29630

tresat commented:
@bot-gradle test RFR

bot-gradle commented:
<!-- {"replyTargetCommentId":4634744761,"triggeredTeamCityBuilds":[{"buildConfigurationId":"Gradle_Master_Check_Stage_ReadyforRelease_Trigger","buildId":"113534163","headRef":"b43dac954977a26e75df8e1e7a9b58bed5e7bfe4"}]} -->
I've triggered the following builds:
- [Ready for Release (Trigger) (Check)](https://builds.gradle.org/build/113534163)
  - [Build Scan](https://ge.gradle.org/scans?search.tags=CI&amp;search.names=gitCommitId,PartOf&amp;search.values=b43dac954977a26e75df8e1e7a9b58bed5e7bfe4,ReadyforRelease&amp;)

bot-gradle commented:
The following builds have failed:
- [Ready for Release (Trigger) (Check)](https://builds.gradle.org/buildConfiguration/Gradle_Master_Check_Stage_ReadyforRelease_Trigger/113534163)
  - [Build Scan](https://ge.gradle.org/scans?search.buildOutcome=failure&amp;search.tags=CI&amp;search.names=gitCommitId&amp;search.values=b43dac954977a26e75df8e1e7a9b58bed5e7bfe4&amp;)

tresat commented:
@bot-gradle test RFR

bot-gradle commented:
<!-- {"replyTargetCommentId":4651668270,"triggeredTeamCityBuilds":[{"buildConfigurationId":"Gradle_Master_Check_Stage_ReadyforRelease_Trigger","buildId":"113609902","headRef":"07a5ecd2cbe1f2ca5dfafa3421d44a4af5e6e17b"}]} -->
I've triggered the following builds:
- [Ready for Release (Trigger) (Check)](https://builds.gradle.org/build/113609902)
  - [Build Scan](https://ge.gradle.org/scans?search.tags=CI&amp;search.names=gitCommitId,PartOf&amp;search.values=07a5ecd2cbe1f2ca5dfafa3421d44a4af5e6e17b,ReadyforRelease&amp;)

bot-gradle commented:
The following builds have failed:
- [Ready for Release (Trigger) (Check)](https://builds.gradle.org/buildConfiguration/Gradle_Master_Check_Stage_ReadyforRelease_Trigger/113609902)
  - [Build Scan](https://ge.gradle.org/scans?search.buildOutcome=failure&amp;search.tags=CI&amp;search.names=gitCommitId&amp;search.values=07a5ecd2cbe1f2ca5dfafa3421d44a4af5e6e17b&amp;)

tresat commented:
@bot-gradle test RFR

bot-gradle commented:
<!-- {"replyTargetCommentId":4653426301,"triggeredTeamCityBuilds":[{"buildConfigurationId":"Gradle_Master_Check_Stage_ReadyforRelease_Trigger","buildId":"113618751","headRef":"59040e4fac4619f97bfa288fef947a29cc4b8c78"}]} -->
I've triggered the following builds:
- [Ready for Release (Trigger) (Check)](https://builds.gradle.org/build/113618751)
  - [Build Scan](https://ge.gradle.org/scans?search.tags=CI&amp;search.names=gitCommitId,PartOf&amp;search.values=59040e4fac4619f97bfa288fef947a29cc4b8c78,ReadyforRelease&amp;)

bot-gradle commented:
The following builds have failed:
- [Ready for Release (Trigger) (Check)](https://builds.gradle.org/buildConfiguration/Gradle_Master_Check_Stage_ReadyforRelease_Trigger/113618751)
  - [Build Scan](https://ge.gradle.org/scans?search.buildOutcome=failure&amp;search.tags=CI&amp;search.names=gitCommitId&amp;search.values=59040e4fac4619f97bfa288fef947a29cc4b8c78&amp;)

tresat commented:
@bot-gradle test this

bot-gradle commented:
<!-- {"replyTargetCommentId":4695488500,"triggeredTeamCityBuilds":[{"buildConfigurationId":"Gradle_Master_Check_Stage_PullRequestFeedback_Trigger","buildId":"113905956","headRef":"c587330f10ec7a0355422614386bd50b49b81f71"}]} -->
I've triggered the following builds:
- [Pull Request Feedback (Trigger) (Check)](https://builds.gradle.org/build/113905956)
  - [Build Scan](https://ge.gradle.org/scans?search.tags=CI&amp;search.names=gitCommitId,PartOf&amp;search.values=c587330f10ec7a0355422614386bd50b49b81f71,PullRequestFeedback&amp;)

bot-gradle commented:
The following builds have failed:
- [Pull Request Feedback (Trigger) (Check)](https://builds.gradle.org/buildConfiguration/Gradle_Master_Check_Stage_PullRequestFeedback_Trigger/113905956)
  - [Build Scan](https://ge.gradle.org/scans?search.buildOutcome=failure&amp;search.tags=CI&amp;search.names=gitCommitId&amp;search.values=c587330f10ec7a0355422614386bd50b49b81f71&amp;)

tresat commented:
@bot-gradle test this

bot-gradle commented:
<!-- {"replyTargetCommentId":4723517864,"triggeredTeamCityBuilds":[{"buildConfigurationId":"Gradle_Master_Check_Stage_PullRequestFeedback_Trigger","buildId":"114053295","headRef":"115306ef71194b39c1f5bfe803e71035acad1bd7"}]} -->
I've triggered the following builds:
- [Pull Request Feedback (Trigger) (Check)](https://builds.gradle.org/build/114053295)
  - [Build Scan](https://ge.gradle.org/scans?search.tags=CI&amp;search.names=gitCommitId,PartOf&amp;search.values=115306ef71194b39c1f5bfe803e71035acad1bd7,PullRequestFeedback&amp;)

bot-gradle commented:
The following builds have failed:
- [Pull Request Feedback (Trigger) (Check)](https://builds.gradle.org/buildConfiguration/Gradle_Master_Check_Stage_PullRequestFeedback_Trigger/114053295)
  - [Build Scan](https://ge.gradle.org/scans?search.buildOutcome=failure&amp;search.tags=CI&amp;search.names=gitCommitId&amp;search.values=115306ef71194b39c1f5bfe803e71035acad1bd7&amp;)

tresat commented:
@bot-gradle test this

bot-gradle commented:
<!-- {"replyTargetCommentId":4735085164,"triggeredTeamCityBuilds":[{"buildConfigurationId":"Gradle_Master_Check_Stage_PullRequestFeedback_Trigger","buildId":"114106224","headRef":"1cca22f18c88315fc3574c84efe21b3947f6db1a"}]} -->
I've triggered the following builds:
- [Pull Request Feedback (Trigger) (Check)](https://builds.gradle.org/build/114106224)
  - [Build Scan](https://ge.gradle.org/scans?search.tags=CI&amp;search.names=gitCommitId,PartOf&amp;search.values=1cca22f18c88315fc3574c84efe21b3947f6db1a,PullRequestFeedback&amp;)

bot-gradle commented:
The following builds have failed:
- [Pull Request Feedback (Trigger) (Check)](https://builds.gradle.org/buildConfiguration/Gradle_Master_Check_Stage_PullRequestFeedback_Trigger/114106224)
  - [Build Scan](https://ge.gradle.org/scans?search.buildOutcome=failure&amp;search.tags=CI&amp;search.names=gitCommitId&amp;search.values=1cca22f18c88315fc3574c84efe21b3947f6db1a&amp;)

tresat committed:
Update comments for accuracy

tresat committed:
Update test expectations

tresat committed:
Revert unnecessary change to test

tresat commented:
@bot-gradle test this

bot-gradle commented:
<!-- {"replyTargetCommentId":4745763782,"triggeredTeamCityBuilds":[{"buildConfigurationId":"Gradle_Master_Check_Stage_PullRequestFeedback_Trigger","buildId":"114146994","headRef":"e22f5fd8f29b74915e17e44a6a0f6520d3047c15"}]} -->
I've triggered the following builds:
- [Pull Request Feedback (Trigger) (Check)](https://builds.gradle.org/build/114146994)
  - [Build Scan](https://ge.gradle.org/scans?search.tags=CI&amp;search.names=gitCommitId,PartOf&amp;search.values=e22f5fd8f29b74915e17e44a6a0f6520d3047c15,PullRequestFeedback&amp;)

tresat commented:
@bot-gradle test RFR

bot-gradle commented:
<!-- {"replyTargetCommentId":4746219092,"triggeredTeamCityBuilds":[{"buildConfigurationId":"Gradle_Master_Check_Stage_ReadyforRelease_Trigger","buildId":"114148507","headRef":"e22f5fd8f29b74915e17e44a6a0f6520d3047c15"}]} -->
I've triggered the following builds:
- [Ready for Release (Trigger) (Check)](https://builds.gradle.org/build/114148507)
  - [Build Scan](https://ge.gradle.org/scans?search.tags=CI&amp;search.names=gitCommitId,PartOf&amp;search.values=e22f5fd8f29b74915e17e44a6a0f6520d3047c15,ReadyforRelease&amp;)

bot-gradle commented:
The following builds have failed:
- [Pull Request Feedback (Trigger) (Check)](https://builds.gradle.org/buildConfiguration/Gradle_Master_Check_Stage_PullRequestFeedback_Trigger/114146994)
  - [Build Scan](https://ge.gradle.org/scans?search.buildOutcome=failure&amp;search.tags=CI&amp;search.names=gitCommitId&amp;search.values=e22f5fd8f29b74915e17e44a6a0f6520d3047c15&amp;)

bot-gradle commented:
The following builds have failed:
- [Ready for Release (Trigger) (Check)](https://builds.gradle.org/buildConfiguration/Gradle_Master_Check_Stage_ReadyforRelease_Trigger/114148507)
  - [Build Scan](https://ge.gradle.org/scans?search.buildOutcome=failure&amp;search.tags=CI&amp;search.names=gitCommitId&amp;search.values=e22f5fd8f29b74915e17e44a6a0f6520d3047c15&amp;)

tresat committed:
Default FallbackVariant for implicit secondary preference

This change implicitly adds the `org.gradle.fallback-variant=false` attribute to consumer requests that do not specify it. This ensures that consumers automatically prefer secondary variants, reducing boilerplate.

It acts as the consumer-side counterpart to the producer-side auto-tagging of primary variants with `fallback-variant=true`, ensuring silent consumers correctly prune primary variants before file resolution begins.

tresat committed:
Adjust variant resolution for fallback variants

Implement a two-pass strategy for variant selection when fallback variants are present. The first pass attempts to match secondary variants by injecting `org.gradle.fallback-variant=false`. If no match or transform is found, a second pass uses the original consumer attributes. This ensures that an "empty" fallback variant can still be chosen as a "nothing to resolve" backstop if no other suitable variant or transform chain is found.

Updated error messages provide more precise details about the attribute matching for `org.gradle.fallback-variant`.

tresat commented:
@bot-gradle test RFR

bot-gradle commented:
<!-- {"replyTargetCommentId":4772369441,"triggeredTeamCityBuilds":[{"buildConfigurationId":"Gradle_Master_Check_Stage_ReadyforRelease_Trigger","buildId":"114268996","headRef":"846f6beb49bf14f563ecd3686e9512c1c4d773f3"}]} -->
I've triggered the following builds:
- [Ready for Release (Trigger) (Check)](https://builds.gradle.org/build/114268996)
  - [Build Scan](https://ge.gradle.org/scans?search.tags=CI&amp;search.names=gitCommitId,PartOf&amp;search.values=846f6beb49bf14f563ecd3686e9512c1c4d773f3,ReadyforRelease&amp;)

bot-gradle commented:
The following builds have failed:
- [Ready for Release (Trigger) (Check)](https://builds.gradle.org/buildConfiguration/Gradle_Master_Check_Stage_ReadyforRelease_Trigger/114268996)
  - [Build Scan](https://ge.gradle.org/scans?search.buildOutcome=failure&amp;search.tags=CI&amp;search.names=gitCommitId&amp;search.values=846f6beb49bf14f563ecd3686e9512c1c4d773f3&amp;)

tresat committed:
Revert smoke test change - behavior should remain identical now

tresat commented:
@bot-gradle test RFR

bot-gradle commented:
<!-- {"replyTargetCommentId":4779370176,"triggeredTeamCityBuilds":[{"buildConfigurationId":"Gradle_Master_Check_Stage_ReadyforRelease_Trigger","buildId":"114301644","headRef":"a9a62f97f1077a28895821322f979106b57ba26b"}]} -->
I've triggered the following builds:
- [Ready for Release (Trigger) (Check)](https://builds.gradle.org/build/114301644)
  - [Build Scan](https://ge.gradle.org/scans?search.tags=CI&amp;search.names=gitCommitId,PartOf&amp;search.values=a9a62f97f1077a28895821322f979106b57ba26b,ReadyforRelease&amp;)

bot-gradle commented:
The following builds have failed:
- [Ready for Release (Trigger) (Check)](https://builds.gradle.org/buildConfiguration/Gradle_Master_Check_Stage_ReadyforRelease_Trigger/114301644)
  - [Build Scan](https://ge.gradle.org/scans?search.buildOutcome=failure&amp;search.tags=CI&amp;search.names=gitCommitId&amp;search.values=a9a62f97f1077a28895821322f979106b57ba26b&amp;)

tresat committed:
Emit deprecation for fallback query path if it is used

tresat committed:
Clarify fallback selection pass deprecation message

Refine the deprecation message to indicate that a "fallback selection pass was required" rather than stating a fallback variant was matched. This clarifies that the deprecation fires upon entering the fallback resolution path, regardless of the final matching outcome. This ensures the warning is always surfaced when this deprecated path is taken, even if the subsequent selection fails.

Update integration tests to expect the refined deprecation messages in relevant scenarios.

tresat committed:
Fix since version to pass sanityCheck

tresat committed:
Fix typo breaking test

tresat commented:
@bot-gradle test RFN

bot-gradle commented:
<!-- {"replyTargetCommentId":4983900569,"triggeredTeamCityBuilds":[{"buildConfigurationId":"Gradle_Master_Check_Stage_ReadyforNightly_Trigger","buildId":"115215771","headRef":"7feeff9926338880d4cb7ad1479e00299bd6a0cf"}]} -->
I've triggered the following builds:
- [Ready for Nightly (Trigger) (Check)](https://builds.gradle.org/build/115215771)
  - [Build Scan](https://ge.gradle.org/scans?search.tags=CI&amp;search.names=gitCommitId,PartOf&amp;search.values=7feeff9926338880d4cb7ad1479e00299bd6a0cf,ReadyforNightly&amp;)

bot-gradle commented:
The following builds have been cancelled:
- [Ready for Nightly (Trigger) (Check)](https://builds.gradle.org/buildConfiguration/Gradle_Master_Check_Stage_ReadyforNightly_Trigger/115215771)
  - [Build Scan](https://ge.gradle.org/scans?search.buildOutcome=failure&amp;search.tags=CI&amp;search.names=gitCommitId&amp;search.values=7feeff9926338880d4cb7ad1479e00299bd6a0cf&amp;)

<details><summary>🔎 7 failure groups (2 build, 5 test)</summary>

**Test failure** · 16 occurrences · [scan](https://ge.gradle.org/s/v5i3zza3siobk)
> java.lang.AssertionError: Standard output line 1 contains an unexpected deprecation warning:
>  - Resolving artifacts for project ':library' required a fallback selection pass because no variant satisfied the request. This behavior has been deprecated. This will fail with an error in Gradle 10. To match a variant tagged 'org.gradle.fallback-variant=true', request 'org.gradle.fallback-variant=true' explicitly on the consumer. Otherwise, expose a variant that carries the attributes this request need…
**Test failure** · 12 occurrences · [scan](https://ge.gradle.org/s/za6mx6u75vwrc)
> java.lang.AssertionError: Standard output line 14 contains an unexpected deprecation warning:
>  - Resolving artifacts for project ':ui-test-hilt-manifest' required a fallback selection pass because no variant satisfied the request. This behavior has been deprecated. This will fail with an error in Gradle 10. To match a variant tagged 'org.gradle.fallback-variant=true', request 'org.gradle.fallback-variant=true' explicitly on the consumer. Otherwise, expose a variant that carries the attributes th…
**Test failure** · 12 occurrences · [scan](https://ge.gradle.org/s/yovakrlubp7ca)
> java.lang.AssertionError: Standard output line 14 contains an unexpected deprecation warning:
>  - Resolving artifacts for project ':ui-test-hilt-manifest' required a fallback selection pass because no variant satisfied the request. This behavior has been deprecated. This will fail with an error in Gradle 10. To match a variant tagged 'org.gradle.fallback-variant=true', request 'org.gradle.fallback-variant=true' explicitly on the consumer. Otherwise, expose a variant that carries the attributes th…
**Test failure** · 8 occurrences · [scan](https://ge.gradle.org/s/xtzzpmd66lgek)
> java.lang.AssertionError: Standard output line 14 contains an unexpected deprecation warning:
>  - Resolving artifacts for project ':ui-test-hilt-manifest' required a fallback selection pass because no variant satisfied the request. This behavior has been deprecated. This will fail with an error in Gradle 10. To match a variant tagged 'org.gradle.fallback-variant=true', request 'org.gradle.fallback-variant=true' explicitly on the consumer. Otherwise, expose a variant that carries the attributes th…
**Build failure** · 1 occurrence · [scan](https://ge.gradle.org/s/sqpin63hgvgdk)
> java.io.IOException: Unable to delete directory 'C:\tcagent1\work\f63322e10dd6b396\testing\performance\build'
>   Failed to delete some children. This might happen because a process has files open or has its working directory set in the target directory.
>   - C:\tcagent1\work\f63322e10dd6b396\testing\performance\build\tmp\buildCommitDistribution…
**Test failure** · 1 occurrence · [scan](https://ge.gradle.org/s/gvs5ppu2btbss)
> org.gradle.integtests.fixtures.executer.UnexpectedBuildFailure: Gradle execution failed in C:\tcagent1\work\f63322e10dd6b396\platforms\core-runtime\wrapper-main\build\tmp\test files\WrapperProp.Test\System_prop._6_6\f5aku with: cmd.exe [/d, /c, "gradlew -i --no-daemon --gradle-user-home "C:\tcagent1\work\f63322e10dd6b396\platforms\core-runtime\wrapper-main\build\tmp\test files\WrapperProp.Test\System_prop._6_6\f5aku\user-home" --warning-mode=all -Dorg.gradle.java.installations.auto-download=fals…
**Build failure** · 1 occurrence · [scan](https://ge.gradle.org/s/roj5zpvlz7amm)
> Only 49 of 50 expected performance test buckets reported results in /home/tcagent1/agent/work/f63322e10dd6b396/perf-results. The aggregate report has been generated from the available data; failing the build so the missing buckets are not silently ignored.
</details>
